lib/bigdecimal.rb

```diff
@@ -3,3 +3,37 @@
 else
   require 'bigdecimal.so'
 end
+
+class BigDecimal
+
+  # Returns the square root of the value.
+  #
+  # Result has at least prec significant digits.
+  #
+  def sqrt(prec)
+    if infinite? == 1
+      exception_mode = BigDecimal.mode(BigDecimal::EXCEPTION_ALL)
+      raise FloatDomainError, "Computation results in 'Infinity'" if exception_mode.anybits?(BigDecimal::EXCEPTION_INFINITY)
```

Comment:
@tompng, how do you think about extracting the infinity check and exception-raising part as a method?

Reply:
This is sqrt-specific condition. For example, `Math.atan(infinity)=π/2`, `"Computation results in 'Infinity'" is not a correct message.

We can re-consider adding a common infinity check method later, for unbounded monotonic increasing function and other types, but I think it's too early for now.